src/backend/gl/src/device.rs

```diff
@@ -213,6 +213,34 @@ impl Device {
             })
     }
 
+    fn specialize_ast(
+        &self,
+        ast: &mut spirv::Ast<glsl::Target>,
+        specializations: &[pso::Specialization],
+    ) {
+        let spec_cons = ast.get_specialization_constants().unwrap();
+        for spec in specializations {
+            let id = spec_cons.clone().iter()
+                .filter(|tspec| tspec.constant_id == spec.id)
+                .next().unwrap().id;
+
+            unsafe {
+                let value = match spec.value {
```

Comment:
Could we just use the same logic from https://github.com/gfx-rs/gfx/blob/master/src/backend/dx12/src/device.rs#L403 here?